src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryMap.cs

```diff
@@ -0,0 +1,314 @@
+                    await mmv.FlushAsync();
+                }
+
+                return contentLength;
+            }
+            catch
+            {
+                // This exception can be triggered by:
+                // 1) Size is greater than the total virtual memory.
+                // 2) Access is invalid for the memory-mapped file.
+                _mapAccessor.Delete(_mapName, _mmf);
+            }
+
+            _logger.LogError($"Cannot put bytes: {_mapName}");
+            return 0;
+        }
+
+        /// <summary>
+        /// Write content from the given <see cref="byte[]"/> into this <see cref="SharedMemoryMap"/>.
+        /// The number of bytes of content being written must be less than or equal to the size of the <see cref="SharedMemoryMap"/>.
+        /// TODO gochaudh:
+        /// What happens when byte[] is larger than 2GB? We throw?
+        /// </summary>
+        /// <param name="content"><see cref="byte[]"/> containing the content to write.</param>
+        /// <returns>Number of bytes of content written.</returns>
+        public Task<long> PutBytesAsync(byte[] content)
+        {
+            using (MemoryStream contentStream = new MemoryStream(content))
+            {
+                return PutStreamAsync(contentStream);
+            }
+        }
+
+        /// <summary>
+        /// Read the content contained in this <see cref="SharedMemoryMap"/> as <see cref="byte[]"/>.
+        /// </summary>
+        /// <returns><see cref="byte[]"/> of content if successful, <see cref="null"/> otherwise.</returns>
+        public async Task<byte[]> GetBytesAsync()
+        {
+            using (MemoryStream contentStream = new MemoryStream())
+            {
+                // TODO gochaudh:
+                // What happens when byte[] is larger than 2GB? We throw after checking contentStream.Length?
+                if (await TryCopyToAsync(0, -1, contentStream))
+                {
+                    return contentStream.ToArray();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Read the content contained in this <see cref="SharedMemoryMap"/> as <see cref="byte[]"/>.
+        /// </summary>
+        /// <param name="offset">Offset to start reading the content from.</param>
+        /// <param name="count">Number of bytes to read.</param>
+        /// <returns><see cref="byte[]"/> of content if successful, <see cref="null"/> otherwise.</returns>
+        public async Task<byte[]> GetBytesAsync(long offset, long count)
+        {
+            using (MemoryStream contentStream = new MemoryStream())
+            {
+                if (await TryCopyToAsync(offset, count, contentStream))
+                {
+                    // TODO gochaudh:
+                    // What happens when byte[] is larger than 2GB? We throw after checking contentStream.Length?
+                    return contentStream.ToArray();
```

Comment:
Using the MemoryStream is convenient but I'm concerned about the memory overhead. Is this stream lazy or are we creating the memory twice?

Reply:
`ToArray()` creates a copy, which I think is fine for now. We can test memory footprint with some sample apps to see what is going on.